### PR TITLE
Financial Reporting - Backend Changes and CSV Generation

### PR DESCRIPTION
**Pull request by akroon3r**
Commit 1:

As a finanace member, a bookings user will require to access an ability to gerenate a report or csv file to help generate a report on monthly/yearly revenues incurred from exams. In order to do this, a new end-point was added to GET all exams based on two url parameteres (start_date and end_date). Based on these parameters, the GET looks at all exams that were booked between those dates (exam and booking model are joined, then queried). Along with this new end-point, this PR includes a new set of postman tests to ensure that this end-point returns exactly what is required in order to generate these reports.

Commit 2:

According to requirements provided to the developers, the ability for a CSR/GA/Liaison to generate a financial report entailed the generation of a CSV file in order to build this report. This report includes fields from the exam, booking, room and offices tables. This file should then be exported directly to the clients browser for them to use. The modifications to the GET request that is used to generate the csv file were tested in postman. Future modifications to this would possibly include the office name or timestamp in the file name of the export csv file.

### CODE REVIEW

File: frontend/src/store/index.js [hunk 1]
```diff
@@ -2082,6 +2098,11 @@ export const store = new Vuex.Store({
       state.exams = payload
     },
 
+    setExamsExport(state, payload){
```
Comment (sjrumsby) [inline frontend/src/store/index.js]: Is this used still?

File: frontend/src/exams/generate-financial-report-modal.vue [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          ...mapMutations([
+            'toggleGenFinReport'
+          ]),
+          submit() {
+
+            console.log("Pressed Submit")
+            let form_start_date = this.startDate
+            let form_end_date = this.endDate
+            let url= '/exams/export/?start_date=' + form_start_date + '&end_date=' + form_end_date
+            this.getExamsExport(url).then(resp => { const FileDownload = require('js-file-download')
```
Comment (sjrumsby) [inline frontend/src/exams/generate-financial-report-modal.vue]: Formatting on this line. `.then` should be newlined.

File: api/postman/bookings-postman.json [hunk 3]
```diff
@@ -2128,6 +2128,7 @@
 			"item": [
 				{
 					"name": "Localhost - Exams Export List",
+<<<<<<< HEAD
```
Comment (sjrumsby) [inline api/postman/bookings-postman.json]: Merge conflicts got committed?

Comment (akroon3r) [reply api/postman/bookings-postman.json]: resolved

File: frontend/src/exams/generate-financial-report-modal.vue (hunk 2 above)
Comment (akroon3r) [reply frontend/src/exams/generate-financial-report-modal.vue]: resolved

File: frontend/src/store/index.js (hunk 1 above)
Comment (akroon3r) [reply frontend/src/store/index.js]: resolved

**Pull request merged**